internal/app/usecase/usecase.go

```diff
@@ -179,3 +211,67 @@ func (au *AppUsecase) GetOrCreateURLs(requestBatchURLs []app.RequestBatchURL) ([
 
 	return responseBatchURLs, nil
 }
+
+func (au *AppUsecase) GetUserURLs(userID uint) ([]app.ResponseUserURL, error) {
+	urls, err := au.AppRepo.GetUserURLs(userID)
+	if err != nil {
+		return nil, err
+	}
+
+	responseUserURLs := []app.ResponseUserURL{}
+	for _, appURL := range urls {
+		responseUserURLs = append(responseUserURLs, app.ResponseUserURL{
+			ShortURL:    au.GenerateShortURL(appURL.ID),
+			OriginalURL: appURL.URL,
+		})
+	}
+
+	return responseUserURLs, nil
+}
+
+func (au *AppUsecase) SendDeleteUserURLsInChan(userID uint, urlIDs []string) error {
+	go func() {
+		for _, urlID := range urlIDs {
+			select {
+			case au.deleteURLsChan <- &app.URL{ID: urlID, UserID: userID}:
+			case <-au.deleteURLsCtx.Done():
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (au *AppUsecase) deleteUserURLs() {
+	logger := loggerInternal.Log
+
+	var urls []*app.URL
+
+	for {
+		select {
+		case appURL := <-au.deleteURLsChan:
+			urls = append(urls, appURL)
+		case <-au.deleteURLsTicker.C:
+			if len(urls) == 0 {
+				continue
+			}
+			logger.Debug("Deleting user URLs",
+				zap.Any("urls", urls),
+			)
+			err := au.AppRepo.DeleteUserURLs(urls)
+			if err != nil {
+				logger.Error("Failed to delete user URLs",
+					zap.Error(err),
+				)
+				continue
+			}
+			urls = nil
```

Comment:
```suggestion
			urls = urls[:0]
```
А можно еще и объявить urls через make с преаллокацией.
https://t.me/bushigo/11
https://t.me/bushigo/12

Reply:
Поправил
Спасибо за канал, подписался)